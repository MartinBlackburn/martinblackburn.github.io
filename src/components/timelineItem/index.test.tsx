// libraries
import { render } from "@testing-library/react";

// component under test
import TimelineItem from "./";

// mocks
jest.mock("../Date", () => () => <div>date component</div>);

describe("TimelineItem component", () => {
    test("renders correctly", () => {
        const { asFragment } = render(
            <TimelineItem title="some title" text="some text" date={new Date("1986-2-11")} />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
