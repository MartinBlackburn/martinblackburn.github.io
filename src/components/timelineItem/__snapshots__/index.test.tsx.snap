// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItem component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="timelineItem"
  >
    <div
      class="timelineItem__line"
    />
    <div
      class="timelineItem__icon"
    />
    <div
      class="timelineItem__content"
    >
      <h2
        class="timelineItem__title"
      >
        some title
      </h2>
      <div>
        date component
      </div>
      <p
        class="timelineItem__text"
      >
        some text
      </p>
    </div>
  </div>
</DocumentFragment>
`;
